test/functional/wallet_encryption.py

```diff
@@ -79,18 +78,18 @@ def run_test(self):
         expected_time = int(time.time()) + MAX_VALUE - 600
         self.nodes[0].walletpassphrase(passphrase2, MAX_VALUE - 600)
         # give buffer for walletpassphrase, since it iterates over all encrypted keys
-        expected_time_with_buffer = time.time() + MAX_VALUE - 600
+        expected_time_with_buffer = int(time.time()) + MAX_VALUE - 600
```

Comment:
This is wrong. From your linked bug report, `expected_time_with_buffer` is `...29.99` and `actual_time` is `...30`.

With `floor/int`, this will give `...29`, which will still fail the assertion.